fontbe/src/hvar.rs

```diff
@@ -0,0 +1,233 @@
+//! Generates an [HVAR](https://learn.microsoft.com/en-us/typography/opentype/spec/HVAR) table.
+
+use std::collections::{BTreeSet, HashMap};
+use std::mem;
+use std::sync::Arc;
+
+use indexmap::IndexMap;
+
+use font_types::MajorMinor;
+use fontdrasil::{
+    coords::NormalizedLocation,
+    orchestration::{Access, Work},
+    types::Axis,
+};
+use fontir::{ir::Glyph, orchestration::WorkId as FeWorkId, variations::VariationModel};
+use write_fonts::{
+    dump_table,
+    tables::{
+        hvar::Hvar,
+        variations::{ivs_builder::VariationStoreBuilder, DeltaSetIndexMap, VariationRegion},
+    },
+    validate::Validate,
+    FontWrite, OtRound,
+};
+
+use crate::{
+    error::Error,
+    orchestration::{AnyWorkId, BeWork, Context, WorkId},
+};
+
+#[derive(Debug)]
+struct HvarWork {}
+
+pub fn create_hvar_work() -> Box<BeWork> {
+    Box::new(HvarWork {})
+}
+
+/// Compute the final size of a table, after it has been serialized to bytes
+fn table_size<T>(table: &T, context: &str) -> Result<usize, Error>
+where
```

Comment:
instead of passing in a string here, you could get the type name with the [`type_name`](http://doc.rust-lang.org/1.73.0/std/any/fn.type_name.html) fn, which would probably be suitable for the purposes of an error message?